File: dev/20190514/session/src/FizzBuzz.js
function generateList() {
  return [...Array(101).keys()].slice(1).map(ix => fizzBuzz(ix));
}

function fizzBuzz(number) {
  if (number % 15 === 0) return "FizzBuzz";
  if (number % 3 === 0) return "Fizz";
  if (number % 5 === 0) return "Buzz";
  return number;
}

module.exports.fizzBuzz = fizzBuzz;
module.exports.generateList = generateList;

File: dev/20190520/k2works/api/src/main/java/k2works/FizzBuzz.java
package k2works;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.IntStream;

public class FizzBuzz {
  public static final int MAX_NUMBER = 100;
  private FizzBuzzValues _values;
  private FizzBuzzType _type;

  public List<FizzBuzzValue> getValues() {
    return this._values.get_values();
  }

  public FizzBuzz(FizzBuzzType type) {
    List<FizzBuzzValue> list = new ArrayList<>();
    _values = new FizzBuzzValues(list);
    this._type = type;
  }

  public FizzBuzzValue generate(int number) {
    return this._type.generate(number);
  }

  public void generateList() {
    IntStream.rangeClosed(1, MAX_NUMBER).forEach(
      number -> _values = this._values.add(this.generate(number))
    );
  }

}

